src/anomalib/models/components/base/anomaly_module.py

```diff
@@ -275,3 +276,65 @@ def on_load_checkpoint(self, checkpoint: dict[str, Any]) -> None:
         """
         self._transform = checkpoint["transform"]
         self.setup("load_checkpoint")
+
+    @classmethod
+    def from_config(
+        cls: type["AnomalyModule"],
+        config_path: str | Path,
+        **kwargs,
+    ) -> "AnomalyModule":
+        """Create a model instance from the configuration.
+
+        Args:
+            config_path (str | Path): Path to the model configuration file.
+            **kwargs (dict): Additional keyword arguments.
+
+        Returns:
+            AnomalyModule: model instance.
+
+        Example:
+            The following example shows how to get model from patchcore.yaml:
+
+            .. code-block:: python
+                >>> model_config = "configs/model/patchcore.yaml"
+                >>> model = AnomalyModule.from_config(config_path=model_config)
+
+            The following example shows overriding the configuration file with additional keyword arguments:
+
+            .. code-block:: python
+                >>> override_kwargs = {"model.pre_trained": False}
+                >>> model = AnomalyModule.from_config(config_path=model_config, **override_kwargs)
+        """
+        from jsonargparse import ActionConfigFile, ArgumentParser
+        from lightning.pytorch import Trainer
+
+        from anomalib import TaskType
+
+        if not Path(config_path).exists():
+            msg = f"Configuration file not found: {config_path}"
+            raise FileNotFoundError(msg)
+
+        model_parser = ArgumentParser()
+        model_parser.add_argument(
+            "-c",
+            "--config",
+            action=ActionConfigFile,
+            help="Path to a configuration file in json or yaml format.",
+        )
+        model_parser.add_subclass_arguments(AnomalyModule, "model", required=False, fail_untyped=False)
+        model_parser.add_argument("--task", type=TaskType | str, default=TaskType.SEGMENTATION)
+        model_parser.add_argument("--metrics.image", type=list[str] | str | None, default=["F1Score", "AUROC"])
+        model_parser.add_argument("--metrics.pixel", type=list[str] | str | None, default=None, required=False)
+        model_parser.add_argument("--metrics.threshold", type=BaseThreshold | str, default="F1AdaptiveThreshold")
+        model_parser.add_class_arguments(Trainer, "trainer", fail_untyped=False, instantiate=False, sub_configs=True)
```

Comment:
Yes I agree with you and I also have a solution for this. 
In OTX, the CLI class is configured to receive the parser as a staticmethod like below.
https://github.com/openvinotoolkit/training_extensions/blob/ce2d99f5551b6f7ede3a6968a6e279c8f8ba185d/src/otx/cli/cli.py#L83-L84
Then we can simply create a parser with a specific purpose as shown below.
https://github.com/openvinotoolkit/training_extensions/blob/ce2d99f5551b6f7ede3a6968a6e279c8f8ba185d/src/otx/cli/utils/jsonargparse.py#L356

However, implementing this now would require more code changes, so i think it would be better to keep it and refactor the CLI later. What do you think?

Reply:
Alright! Let's refactor this later.